RebelAllianceBank/Classes/FileHandler.cs

```diff
@@ -0,0 +1,184 @@
+                    while ((read = sr.ReadLine()) != null)
+                    {
+                        string[] row = read.Split('-');
+                        T instance = StoredData(row);
+                        if (instance != null)
+                        {
+                            savedList.Add(instance);
+                        }
+                    }
+                }
+            }
+            return savedList;
+        }
+        /// <summary>
+        /// Converts a string array to an <see cref="IUser"/> object.
+        /// </summary>
+        /// <param name="row">A string array representing a user row from the file.</param>
+        /// <returns>An <see cref="IUser"/> object or null if the row is invalid.</returns>
+        public IUser StoredUser(string[] row)
+        {
+            switch (row[3])
+            {
+                case "true":
+                    return new Admin
+                    {
+                        ID = Convert.ToInt16(row[0]), // unique id
+                        PersonalNum = row[1], // 8802252525
+                        Password = row[2],
+                        Surname = row[3],
+                        Lastname = row[4],
+                        //IsAdmin = bool.Parse(row[3]),
+                    };
+                case "false":
+                    return new Customer
+                    {
+                        ID = Convert.ToInt16(row[0]),
+                        PersonalNum = row[2],
+                        Password = row[2],
+                        Surname = row[3],
+                        Lastname = row[4],
+                        //IsAdmin = bool.Parse(row[3]),
+                    };
+                default:
+                    return null;
+            }
+        }
+        /// <summary>
+        /// Writes a user to the file and returns the created <see cref="IUser"/> object.
+        /// </summary>
+        /// <param name="id">The ID of the user.</param>
+        /// <param name="un">The username of the user.</param>
+        /// <param name="pw">The password of the user.</param>
+        /// <param name="isAdmin">Specifies whether the user is an admin or not.</param>
+        /// <returns>The created <see cref="IUser"/> object or null if writing failed.</returns>
+        public IUser WriteUserToFile(int id, string un, string pw, bool isAdmin)
+        {
+            string filePath = "..\\..\\..\\users.txt";
+            if (File.Exists(filePath))
+            {
+                using (StreamWriter sw = new StreamWriter(filePath, append: true))
+                {
+                    string write = $"{id}-{un}-{pw}-{isAdmin}";
+                    sw.WriteLine(write, Environment.NewLine);
+                }
+            }
+            IUser user = StoredUser(new string[] { id.ToString(), un, pw, isAdmin.ToString().ToLower() });
+            if (user != null)
+            {
+                return user;
+            }
+            return null;
+        }
```

Comment:
Why does it return an IUser? Shouldn't it just save to file and not return anything?

Reply:
Had to do it on how i built the code to be able to renew the user list. Will probably be changed on newer revision